evaluate/repository.go

```diff
@@ -77,3 +78,56 @@ func Repository(logger *log.Logger, resultPath string, model evalmodel.Model, la
 
 	return repositoryAssessment, problems, nil
 }
+
+// TemporaryRepository creates a temporary repository and initializes a git repo in it.
+func TemporaryRepository(testDataPath string, repositoryPath string) (temporaryRepositoryPath string, err error) {
```

Comment:
Why do we have `testDataPath` and `repositoryPath` separate? We concat them anyways into `dataPath`  within this function.